mlflow/protos/model_registry.proto

```diff
@@ -358,6 +372,35 @@ message ListRegisteredModels {
   }
 }
 
+message SearchRegisteredModels {
+  option (scalapb.message).extends = "com.databricks.rpc.RPC[$this.Response]";
+
+  // String filter condition, like "name LIKE 'my-model-name'".
+  // Interpreted in the backend automatically as "name LIKE '%my-model-name%'".
+  // Single boolean condition, with string values wrapped in single quotes.
+  optional string filter = 1;
+
+  // Maximum number of models desired. Default is 100. Max threshold is 1000.
+  optional int64 max_results = 2 [default = 100];
+
+  // List of columns to be ordered by including model name, version, stage with an
+  // optional "DESC" or "ASC" annotation, where "ASC" is the default.
+  // Tiebreaks are done by latest stage transition timestamp, followed by name ASC, followed by
+  // version DESC.
+  repeated string order_by = 3;
+
+  // Pagination token to go to the next page based on a previous search query.
+  optional string page_token = 4;
```

Comment:
slight nit: Since we don't use the `order_by` in the backend implementation, it might make sense to put `page_token` first.

Reply:
`order_by` is not implemented in SQL store yet (currently raises error) but different backends may support that.